sdk/eventgrid/Azure.Messaging.EventGrid/tests/ConsumeEventTests.cs

```diff
@@ -1212,17 +1167,43 @@ public void ConsumeResourceActionCancelEvent()
             EventGridEvent[] events = EventGridEvent.ParseMany(new BinaryData(requestContent));
 
             Assert.NotNull(events);
-            Assert.True(events[0].TryGetSystemEventData(out object eventData));
-            Assert.AreEqual("72f988bf-86f1-41af-91ab-2d7cd011db47", (eventData as ResourceActionCancelEventData).TenantId);
+            Assert.True(events[0].TryGetSystemEventData(out object data));
+            var eventData = data as ResourceActionCancelEventData;
 
-            var authorizationJson = JsonDocument.Parse((eventData as ResourceActionCancelEventData).Authorization).RootElement;
-            Assert.AreEqual("/subscriptions/sub/resourceGroups/rg/providers/Microsoft.Web/sites/function/host/default", authorizationJson.GetProperty("scope").GetString());
+            AssertResourceEventData(eventData);
+        }
+
+        // Using dynamic to avoid duplicating the test cases for each event. The events don't share a common base type but they all have the
+        // properties being tested below.
+        private static void AssertResourceEventData(dynamic eventData)
+        {
```

Comment:
Since you're doing an `as` on the data before calling this, should we consider starting off with a null check, since `eventData` could potentially have failed to convert?